sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/CustomAnalyzerSyncTests.java

```diff
@@ -79,33 +82,32 @@
 import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.Test;
 
+import static com.azure.search.documents.TestHelpers.assertHttpResponseException;
 import static com.azure.search.documents.TestHelpers.assertObjectEquals;
+import static com.azure.search.documents.TestHelpers.generateRequestOptions;
+import static com.azure.search.documents.TestHelpers.waitForIndexing;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 
-public class CustomAnalyzerSyncTests extends SearchServiceTestBase {
+public class CustomAnalyzerSyncTests extends SearchTestBase {
     private static final String NAME_PREFIX = "azsmnet";
+    private static final Collection<CharFilterName> CHAR_FILTER_NAMES = new ArrayList<>(CharFilterName.values());
 
     private SearchServiceClient searchServiceClient;
-    private static Collection<CharFilterName> charFilterNames;
-
-    static {
-        getAllCharFilterName();
-    }
+    private final List<String> indexesToCleanup = new ArrayList<>();
 
     @Override
     protected void beforeTest() {
         super.beforeTest();
         searchServiceClient = getSearchServiceClientBuilder().buildClient();
     }
 
-    private static void getAllCharFilterName() {
-        charFilterNames = new ArrayList<>();
-        for (CharFilterName name : CharFilterName.values()) {
-            charFilterNames.add(name);
+    @Override
+    protected void afterTest() {
+        super.afterTest();
+        for (String index : indexesToCleanup) {
```

Comment:
Is it better to clean up the indexesToCleanup after test or re-initiate before run?

Reply:
I think it would be better to cleanup after the test, this makes it so the test itself maintains all the logic it uses.